### ISSUE TITLE
[BUG] `AssertionError` for `test_arsenal.py::test_contracted_arsenal`, `len(obj.estimators_)` is not more than 1

### ISSUE THREAD

**Issue by achieveordie** [sktime/sktime#5488]
**Describe the bug**

First noticed in the CI runs of #5486 where multiple workers failed because of this error ([1](https://github.com/sktime/sktime/actions/runs/6635472949/job/18026451921?pr=5486), [2](https://github.com/sktime/sktime/actions/runs/6628194077/job/18006354777#step:3:588), [3](https://github.com/sktime/sktime/actions/runs/6628194077/job/18006357748#step:3:680)).
**To Reproduce**

While I'm not familiar with the inner workings of the `Arsenal` ensemble, I think this error is arising for the assertion statement in `/classification/kernel_based/tests/test_arsenal.py`:
```python
assert len(arsenal.estimators_) > 1
```
Which should have been more than or equal to 1 since we can have an 'ensemble' containing only 1 child estimator. But I could be wrong and these errors might be originating for sporadic reasons. Note that these files have not been changed recently so it is unlikely to be coming from any recent changes.

**Expected behavior**

**Additional context**

**Versions**
<details>

</details>

**Comment by fkiraly** [comment]
Yes, an ensemble of 1 is valid, what we should quickly check if it is also possible that the logic results in an ensemble of zero, and this results in an error.

Not handling the case of 1 or 0 estimators is a common "mistake" in implementation of bootstrap ensembles.

What I typically do is allow 1 and default to the `DummyClassifier` logic (not the estimator but the logic) in the case of 0.
